packages/mobile/src/harmony-native/components/TextLink/hooks.ts

```diff
@@ -0,0 +1,92 @@
+import { useCallback } from 'react'
+
+import { useLeavingAudiusModal } from '@audius/common/store'
+import {
+  getPathFromAudiusUrl,
+  isAllowedExternalLink
+} from '@audius/common/utils'
+import { useLinkProps, useLinkTo } from '@react-navigation/native'
+import type { GestureResponderEvent } from 'react-native'
+import { Linking } from 'react-native'
+
+import { useToast } from 'app/hooks/useToast'
+
+import type { ExternalLinkProps, InternalLinkToProps } from './types'
+
+const messages = {
+  error: 'Unable to open this URL'
+}
+
+export const useHandlePressExternalUrl = (props: ExternalLinkProps) => {
```

Comment:
just double checking that we dont want to do component wrapper style which each of those rendering just the hook it needs?

Reply:
Without the extra Text wrapper the color animation doesn't apply to UserLinks that aren't nested inside a Text component. I'm actually not sure why that's the case

Reply:
Yeah you're probably right, I was thinking about that. basically going full circle to what you had already 😂